### PR TITLE
RJS-2771: Upgrade to Realm Core v14.4.0

### LINKED ISSUE #6568: Upgrade to Realm Core 14.4.0

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes #6568, #6554, and #6558

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

**Review by kraenhansen — APPROVED** [review]

**Review by elle-j — APPROVED** [review]

File: packages/realm/bindgen/src/templates/jsi.ts [hunk 1]
```diff
@@ -239,6 +239,9 @@ function convertPrimToJsi(addon: JsiAddon, type: string, expr: string): string {
     case "uint64_t":
       return `bigIntFromU64(_env, ${expr})`;
 
+    case "std::chrono::milliseconds":
+      return `bigIntFromU64(_env, std::chrono::milliseconds(${expr}ms).count())`;
```
Comment (kraenhansen) [inline packages/realm/bindgen/src/templates/jsi.ts:243]:
```suggestion
      return `bigIntFromU64(_env, std::chrono::milliseconds(${expr}).count())`;
```

File: packages/realm/bindgen/src/templates/node.ts [hunk 2]
```diff
@@ -191,6 +191,9 @@ function convertPrimToNode(addon: NodeAddon, type: string, expr: string): string
     case "uint64_t":
       return `Napi::BigInt::New(${env}, ${expr})`;
 
+    case "std::chrono::milliseconds":
+      return `Napi::Number::New(${env}, std::chrono::milliseconds(${expr}ms).count())`;
```
Comment (kraenhansen) [inline packages/realm/bindgen/src/templates/node.ts:195]:
```suggestion
      return `Napi::Number::New(${env}, std::chrono::milliseconds(${expr}).count())`;
```

File: packages/realm/bindgen/src/templates/jsi.ts [hunk 3]
```diff
@@ -333,6 +336,9 @@ function convertPrimFromJsi(addon: JsiAddon, type: string, expr: string): string
     case "uint64_t":
       return `bigIntToU64(_env, jsi::Value(_env, ${expr}))`;
 
+    case "std::chrono::milliseconds":
+      return `std::chrono::milliseconds(bigIntToU64(_env, jsi::Value(_env, ${expr})))`;
```
Comment (kraenhansen) [inline packages/realm/bindgen/src/templates/jsi.ts:340]: Thinking more about this change. We should have been able to ignore this primitive all together, by simply not including it in the `js_opt_in_spec`. Did bindgen fail if you didn't put this code in?

File: packages/realm/bindgen/src/templates/jsi.ts [hunk 4]
```diff
@@ -239,6 +239,9 @@ function convertPrimToJsi(addon: JsiAddon, type: string, expr: string): string {
     case "uint64_t":
       return `bigIntFromU64(_env, ${expr})`;
 
+    case "std::chrono::milliseconds":
+      return `bigIntFromU64(_env, std::chrono::milliseconds(${expr}).count())`;
```
Comment (kraenhansen) [inline packages/realm/bindgen/src/templates/jsi.ts:243]: Thinking more about this change. We should have been able to ignore this primitive all together, by simply not including it in the `js_opt_in_spec.yml`. Did bindgen fail if you didn't put this code in? If not - perhaps we should extend the "opt-in" feature to include primitives? What do you think @elle-j?

**Pull request merged**